### ISSUE TITLE
Automapper 6.1.1 throws error mapping non-null nullable to nullable property.

### ISSUE THREAD

**Issue by wdupree** [AutoMapper/AutoMapper#2203]
### Source/destination types
Source Type: DeductibleDto
```csharp
public class DeductibleDto
    {
        public virtual CoverageLevelType CoverageLevelType { get; set; }
        public decimal? Maximum { get; set; }
        public decimal? Remaining { get; set; }
        public DeductibleType DeductibleType { get; set; }
    }
```
Destination Type: Deductible
```csharp
public class Deductible : CostsBase
    {
        public virtual CoverageLevelTypeEnum CoverageLevelType { get; set; }
        public decimal? Maximum { get; set; }
        public decimal? Remaining { get; set; }
        public DeductibleTypeEnum DeductibleType { get; set; }
    }
```
### Mapping configuration

```csharp
public Deductible Map(DeductibleDto input)
        {
            return Mapper.Map<Deductible>(input.ThrowIfNull("input"));
        }
```

### Expected behavior

The mapper should pass the value from the DTO to the local object. This works as expected in 6.0.2 and earlier.

### Actual behavior

Exception: System.PlatformNotSupportedException
Message: The expression 'Equal' is not supported for type 'System.Nullable'1[System.Decimal]'
StackTrace:
   at System.Linq.Expressions.Interpreter.EqualInstruction.Create(Type type, Boolean liftedToNull)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileEqual(Expression left, Expression right, Boolean liftedToNull)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBinaryExpression(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.Compile(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileConditionalExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.Compile(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileVariableAssignment(BinaryExpression node, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileAsVoid(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockStart(BlockExpression node)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileAsVoid(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileAsVoid(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.Compile(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileTryExpression(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileAsVoid(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockStart(BlockExpression node)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.Compile(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileConditionalExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.Compile(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileBlockExpression(Expression expr, Boolean asVoid)
   at System.Linq.Expressions.Interpreter.LightCompiler.Compile(Expression expr)
   at System.Linq.Expressions.Interpreter.LightCompiler.CompileTop(LambdaExpression node)
   at System.Linq.Expressions.LambdaExpression.Compile()
   at AutoMapper.MapperConfiguration.CreateMapperFuncs(MapRequest mapRequest)
   at AutoMapper.LockingConcurrentDictionary`2.<>c__DisplayClass2_1.<.ctor>b__1()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at AutoMapper.MapperConfiguration.GetUntypedMapperFunc(MapRequest mapRequest)
   at AutoMapper.Mapper.AutoMapper.IMapper.Map[TDestination](Object source)
   at BCBSLA_Proto.Mappers.Costs.DeductibleMapper.Map(DeductibleDto input)
   at BCBSLA_Proto.Mappers.Costs.DeductibleBaseMapper.Map(Cost input)
   at BCBSLA_Proto.Factories.CostsFactory.Create(CostResponse input)
   at BCBSLA_Proto.Builders.CostsBuilder.BuildCostsPanels(CostResponse costResponse)
   at BCBSLA_Proto.Builders.CostsBuilder.BuildCostsQuickViewPanel(CostResponse costResponse)
   at BCBSLA_Proto.Views.LandingPage.<GetCostListing>d__17.MoveNext()

**Comment by wdupree** [comment]
FYI, also seems to fail for double? and DateTime? types.

**Comment by lbargaoanu** [comment]
A repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by wdupree** [comment]
Ok... I tried to create a sample console app that mapped the decimal properties just like the UWP app this is in and (of course) it works perfectly. Even stranger - if I do this:

```csharp
            Mapper.Initialize(m =>
            {
                //manual mappings removed for brevity
                m.CreateMap<decimal?, decimal?>().ConvertUsing<NullDecimalPassThrough>();
                m.CreateMap<DateTime?, DateTime?>().ConvertUsing<NullDateTimePassThrough>();
                m.CreateMissingTypeMaps = true;
            });

            Mapper.AssertConfigurationIsValid();
```
```csharp
    internal class NullDecimalPassThrough : ITypeConverter<decimal?, decimal?>
    {
        public decimal? Convert(decimal? source, decimal? destination, ResolutionContext context)
        {
            return source;
        }
    }

    internal class NullDateTimePassThrough : ITypeConverter<DateTime?, DateTime?>
    {
        public DateTime? Convert(DateTime? source, DateTime? destination, ResolutionContext context)
        {
            return source;
        }
    }
```

The mapper works perfectly fine... could it be something that's part of the UWP/.NET Core/etc. that's causing problems? I'll try and create a basic UWP later and see if I can get it to break.

**Comment by lbargaoanu** [comment]
Yes, that's why I asked for a repro :) Most likely the problem is not AM.

**Comment by SamuelDebruyn** [comment]
My UWP app does the same, exception on nullable decimals and datetimes.

**Comment by lbargaoanu** [comment]
If your stack trace is similar to the one above, there's nothing we can do about it.

**Comment by SamuelDebruyn** [comment]
Yes, it's the same. Something I can do as a workaround except for not using nullable value types?

**Comment by lbargaoanu** [comment]
Try the [MyGet](https://github.com/AutoMapper/AutoMapper/wiki/The-MyGet-Build) build. If that fails, maybe ask in a .NET Core forum.

**Comment by SamuelDebruyn** [comment]
For anyone else with the same issue: the latest MyGet build doesn't work.
The latest 5.x version doesn't have this issue.

**Comment by jbogard** [comment]
Any ideas on how to repro this with a test or similar?

**Comment by premiaware** [comment]
I have the same problem with UWP project, the version 5.2.0 working well

**Comment by alesdvorakcz** [comment]
Same problem with UWP here. Downgrading Automapper to 5.2 works

**Comment by lbargaoanu** [comment]
So nobody reported this in a .NET Core forum. I guess it can wait.

**Comment by premiaware** [comment]
i have try to upload AutoMapper to Last Version but the problem is the same in UWP project, exception
"The expression 'Equal' is not supported for nullable type 'System.Nullable`1[System.DateTimeOffset]'"
How can fix the problem? with 5.2 version working well, but have other problem if i compile in release mode.
